File: backend/src/routes/user_routes.rs
use std::sync::Arc;

use axum::{routing::get, Router};

use crate::{handlers::image_handler, handlers::user_handler, state::AppState};

pub fn routes() -> Router<Arc<AppState>> {
    Router::new()
        .route(
            "/",
            get(user_handler::list_users).post(user_handler::add_user),
        )
        .route("/{id}/image", get(image_handler::get_user_image))
        .route(
            "/{id}",
            get(user_handler::fetch_user)
                .put(user_handler::alter_user)
                .delete(user_handler::remove_user),
        )
}

File: src/api/userRoutes.js

import { apiClient } from './apiClient';

export const user_routes = {
  
  getAllUsers: () => {
    return apiClient.request(`/api/user`,`GET`); 
  },

  addUser: (data) => {
    return apiClient.request(`/api/user`,`POST`,data); 
  },
  
  fetchUser: (id) => {
    return apiClient.request(`/api/user/${id}`,`GET`);
  },

  alterUser: (id,data) => {
    return apiClient.request(`/api/user/${id}`,`PUT`,data);
  },

  removeUser: (id) => {
    return apiClient.request(`/api/user/${id}`,`DELETE`);
  }
};